geode-core/src/main/java/org/apache/geode/management/internal/cli/functions/RegionAttributesWrapper.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package org.apache.geode.management.internal.cli.functions;
+
+import java.io.Serializable;
+import java.util.List;
+
+import org.apache.geode.cache.CacheListener;
+import org.apache.geode.cache.CacheLoader;
+import org.apache.geode.cache.CacheWriter;
+import org.apache.geode.cache.RegionAttributes;
+import org.apache.geode.management.internal.cli.domain.ClassName;
+
+/**
+ * This class extracts serializable components from RegionAttributes and represents them simply as
+ * strings. The relevant RegionAttributes fields are nulled out so that Serialzation can occur where
+ * the relevant callbacks do not have classes defined. For example when retrieving a RegionAttribute
+ * from a server to a locator.
+ */
+public class RegionAttributesWrapper<K, V> implements Serializable {
```

Comment:
introducing a new Wrapper class is probably easier to implement based on the current code base, but introducing another layer of logic to this flow that is already very complicated. Have you tried having the FetchRegionAttributesFunction return a RegionFunctionArgs instead of the RegionAttributes? I think it boils down to adding a constructor in RegionFunctionArgs given the RegionAttributes. To me, this is much easier to understand.